### PR TITLE
Many to many tag place

### PR DESCRIPTION
**Pull request by lennart-sakowsky**
Now possible: 
saving a new tag related to a (new) place,
saving multiple tags for one place,
saving the same tag for multiple places.
Please have a look at the TagController, the TagSerializer and the PlaceSerializer.
(The create method in the TagController will most likely go.)

### CODE REVIEW

File: mapit-backend/src/Controller/TagController.php [hunk 1]
```diff
@@ -31,6 +35,49 @@ public function index(TagRepository $tagRepository, TagSerializer $tagSerializer
...
+        ]);
+
+        if(!($tagExists)) {
+            $tag = new Tag();
+            $tag->setName($postData->getName());
+            $tagExists = $tag;
+        } 
+        
+        $tag = $tagExists;
+
```
Comment (tine-krueger) [inline mapit-backend/src/Controller/TagController.php:65]: you could name the variable in line 54 $tag directly.

File: mapit-backend/src/Serializer/PlaceSerializer.php [hunk 2]
```diff
@@ -34,15 +34,13 @@ public function serialize($attributes) {
...
-        $placeObject->setName($postData->name);
-        $placeObject->setStreet($postData->street);
-        $placeObject->setZipcode($postData->zipcode);
-        $placeObject->setLatitude($postData->latitude);
-        $placeObject->setLongitude($postData->longitude);
+        $placeObject->setName($postData->getName());
+        $placeObject->setStreet($postData->getStreet());
+        $placeObject->setZipcode($postData->getZipcode());
+        $placeObject->setLatitude($postData->getLatitude());
+        $placeObject->setLongitude($postData->getLongitude());
```
Comment (tine-krueger) [inline mapit-backend/src/Serializer/PlaceSerializer.php:43]: Question: you create a new Place, when related to a tag. Does it make sense to add the tag right here?

File: mapit-backend/src/Serializer/TagSerializer.php [hunk 3]
```diff
@@ -36,6 +37,16 @@ public function deserialize($content) {
...
+        $taggedPlace = $postData->taggedPlace;
+        
+        $placeObject = new Place();
+        $placeObject->setName($taggedPlace->name);
+        $placeObject->setStreet($taggedPlace->street);
+        $placeObject->setZipcode($taggedPlace->zipcode);
+        $placeObject->setLatitude($taggedPlace->latitude);
+        $placeObject->setLongitude($taggedPlace->longitude);
+        $tagObject->addPlace($placeObject);
+
```
Comment (tine-krueger) [inline mapit-backend/src/Serializer/TagSerializer.php:49]: Looks good.

File: mapit-backend/src/Controller/TagController.php (hunk 1 above)
Comment (lennart-sakowsky) [reply mapit-backend/src/Controller/TagController.php:65]: Fair point – I will do the same with $placeExists.

**Pull request merged**